core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/progress/listener/DeafultProgressUpdater.java

```diff
@@ -0,0 +1,175 @@
+import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.core.SdkResponse;
+import software.amazon.awssdk.core.internal.progress.ProgressListenerContext;
+import software.amazon.awssdk.core.internal.progress.ProgressListenerFailedContext;
+import software.amazon.awssdk.core.internal.progress.snapshot.DefaultProgressSnapshot;
+import software.amazon.awssdk.core.progress.listener.SdkExchangeProgress;
+import software.amazon.awssdk.core.progress.snapshot.ProgressSnapshot;
+import software.amazon.awssdk.http.SdkHttpRequest;
+
+/**
+ * ProgressUpdater exposes methods that invokes listener methods to update and store request progress state
+ */
+@SdkInternalApi
+public class DeafultProgressUpdater implements ProgressUpdater {
+    private final DefaultSdkExchangeProgress requestBodyProgress;
+    private final DefaultSdkExchangeProgress responseBodyProgress;
+    private ProgressListenerContext context;
+    private final ProgressListenerInvoker listenerInvoker;
+
+    public DeafultProgressUpdater(SdkRequest sdkRequest,
+                                  Long requestContentLength) {
+        DefaultProgressSnapshot.Builder uploadProgressSnapshotBuilder = DefaultProgressSnapshot.builder();
+        uploadProgressSnapshotBuilder.transferredBytes(0L);
+        Optional.ofNullable(requestContentLength).ifPresent(uploadProgressSnapshotBuilder::totalBytes);
+
+        ProgressSnapshot uploadProgressSnapshot = uploadProgressSnapshotBuilder.build();
+        requestBodyProgress = new DefaultSdkExchangeProgress(uploadProgressSnapshot);
+
+        DefaultProgressSnapshot.Builder downloadProgressSnapshotBuilder = DefaultProgressSnapshot.builder();
+        downloadProgressSnapshotBuilder.transferredBytes(0L);
+        ProgressSnapshot downloadProgressSnapshot = downloadProgressSnapshotBuilder.build();
+        responseBodyProgress = new DefaultSdkExchangeProgress(downloadProgressSnapshot);
+
+        context = ProgressListenerContext.builder()
+                                         .request(sdkRequest)
+                                         .uploadProgressSnapshot(uploadProgressSnapshot)
+                                         .downloadProgressSnapshot(downloadProgressSnapshot)
+                                         .build();
+
+        listenerInvoker = new ProgressListenerInvoker(sdkRequest.overrideConfiguration()
+                                                                .map(RequestOverrideConfiguration::progressListeners)
+                                                                .orElse(Collections.emptyList()));
+    }
+
+    @Override
+    public void updateRequestContentLength(Long requestContentLength) {
+        requestBodyProgress.updateAndGet(b -> b.totalBytes(requestContentLength));
+    }
+
+    @Override
+    public void updateResponseContentLength(Long responseContentLength) {
+        responseBodyProgress.updateAndGet(b -> b.totalBytes(responseContentLength));
```

Comment:
Same here, appear to be unused methods

Reply:
This is used here : https://github.com/aws/aws-sdk-java-v2/pull/5044/files/f6f2c251c285764a64a733cfb18e6bb2a2eae686#diff-55b86f9d7514135beb5728419048ce5e83255d1967a6dfe5dfc60a2bb09e650eR80